prometheus-openstack-exporter

```diff
@@ -1143,7 +1143,4 @@ if __name__ == "__main__":
         data_gatherer = DataGatherer()
         data_gatherer.start()
     server = HTTPServer(("", config.get("listen_port")), handler)
-    server.timeout = config.get("cache_refresh_interval", 900)
-    while True:
-        sleep(0)
-        server.handle_request()
+    server.serve_forever()
```

Comment:
Yeah, I removed this intentionally as it was only added as part of a previously bad fix in https://github.com/canonical/prometheus-openstack-exporter/commit/546ece6cdd4fd9dadf8802392bbdae0fc8e2993b

The intent seems to have been to bail out of any stalled serving connections so that we could then yield to the DataGatherer.

The refresh_interval itself is still respected by the `sleep(self.refresh_interval)` on line 296.